state/in_memory_store_test.go

```diff
@@ -48,31 +48,65 @@ func Test_NewInMemoryStore(t *testing.T) {
 			assert.Equal(t, tc.expected, *actual)
 		})
 	}
+
+	t.Run("stored config is dereferenced", func(t *testing.T) {
+		finalizedConf := config.DefaultConfig()
+		finalizedConf.Finalize()
+		actual := NewInMemoryStore(finalizedConf)
+
+		// Confirm that input and stored config have same values
+		assert.Equal(t, *finalizedConf, *actual.conf.conf)
+
+		// Confrm that input and stored config reference different objects
+		assert.NotSame(t, finalizedConf, actual.conf.conf)
+
+		// Confrm that input and stored config fields reference different objects
+		assert.NotSame(t, finalizedConf.Tasks, actual.conf.conf.Tasks)
+	})
 }
 
 func Test_InMemoryStore_GetConfig(t *testing.T) {
```

Comment:
would it make sense to combine the tests from the table with the t.Run() test?

It seems like the second test (the t.Run() test) tests the first test scenario and more

Reply:
same with the above test? I don't think the extra checks add a lot of overhead

Reply:
If I understand you correctly, I think you're suggesting to combine the separated dereference test-case with the table test?

If so, I think it is possible. As you mentioned, it would require some extra handling for the some test cases. I'm also planning to add more read methods which will also require extra handling:
 - [example GetTask test](https://github.com/hashicorp/consul-terraform-sync/pull/794/files#diff-646d14a48a3be7e665abc90317bb268d1064583c15147b61fd7c4236c2f96754R263-R273) would require an extra if-statement
 - [example of GetAllTask test](https://github.com/hashicorp/consul-terraform-sync/pull/794/files#diff-646d14a48a3be7e665abc90317bb268d1064583c15147b61fd7c4236c2f96754R141-R145) also requires an extra if statement
 - [example of GetConfig test](https://github.com/hashicorp/consul-terraform-sync/pull/794/files#diff-646d14a48a3be7e665abc90317bb268d1064583c15147b61fd7c4236c2f96754L70-R100) doesn't require an extra handling

I ended up making it outside the table-test so that it would be consistent across tests and easier to identify.

Reply:
ok, sure, with the extensions it makes sense to leave it separate, I'm okay with this :)